openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +116,267 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+            (chapter1.location, timedelta(days=14)),
+            (sequential1.location, {'due': timedelta(days=14)}),
+            (vertical1.location, {'due': timedelta(days=14)}),
+            (problem1.location, {'due': timedelta(days=14)}),
+            (chapter2.location, timedelta(days=28)),
+            (sequential2.location, {'due': timedelta(days=28)}),
+            (vertical2.location, {'due': timedelta(days=28)}),
+            (problem2.location, {'due': timedelta(days=28)})
+        ]
+        course = modulestore().get_item(course.location)
+        self.assertCountEqual(extract_dates_from_course(course), expected_dates)
+
+        # course 4: a section with a subsection that has due_num_weeks and multiple sections without due_num_weeks that has graded content.
+        # Current PLS should apply for the subsections without due_num_weeks that has graded content.
+        course = self.courses[3]
+        chapter1 = course.get_children()[0]
+        sequential1 = chapter1.get_children()[0]
+        vertical1 = sequential1.get_children()[0]
+        problem1 = vertical1.get_children()[0]
+
+        chapter2 = course.get_children()[1]
+        sequential2 = chapter2.get_children()[0]
+        vertical2 = sequential2.get_children()[0]
+        problem2 = vertical2.get_children()[0]
+
+        chapter3 = course.get_children()[2]
+        sequential3 = chapter3.get_children()[0]
+        vertical3 = sequential3.get_children()[0]
+        problem3 = vertical3.get_children()[0]
+
+        chapter4 = course.get_children()[3]
+        sequential4 = chapter4.get_children()[0]
+        vertical4 = sequential4.get_children()[0]
+        problem4 = vertical4.get_children()[0]
+
+        expected_dates = [
+            (course.location, {}),
+            (chapter1.location, timedelta(days=14)),
+            (sequential1.location, {'due': timedelta(days=14)}),
+            (vertical1.location, {'due': timedelta(days=14)}),
+            (problem1.location, {'due': timedelta(days=14)}),
+            (chapter2.location, timedelta(days=28)),
+            (sequential2.location, {'due': timedelta(days=28)}),
+            (vertical2.location, {'due': timedelta(days=28)}),
+            (problem2.location, {'due': timedelta(days=28)}),
+            (chapter3.location, timedelta(days=42)),
+            (sequential3.location, {'due': timedelta(days=42)}),
+            (vertical3.location, {'due': timedelta(days=42)}),
+            (problem3.location, {'due': timedelta(days=42)}),
+            (chapter4.location, timedelta(days=56)),
+            (sequential4.location, {'due': timedelta(days=56)}),
+            (vertical4.location, {'due': timedelta(days=56)}),
+            (problem4.location, {'due': timedelta(days=56)})
+        ]
+        course = modulestore().get_item(course.location)
+        with patch.object(utils, 'get_expected_duration', return_value=timedelta(weeks=8)):
+            actual = extract_dates_from_course(course)
```

Comment:
Same as the above scenario, make this more fun and explicit. It follows the default logic presently and makes it hard to see your work in action